internal/crud/crud.go

```diff
@@ -1,4 +1,4 @@
-// Copyright 2020-2022 the Pinniped contributors. All Rights Reserved.
+// Copyright 2020-2024 the Pinniped contributors. All Rights Reserved.
```

Comment:
I see a lot of diffs in this PR related to CRUD and storage and it's not clear to me why this is related to ID token lifetimes. Is this because we want the K8s object representing the issued token to last no longer than the token itself (which is not a dynamically computed value), or something along those lines?

Reply:
This is mostly because I wanted to bump the version number of all the session storage types. Whenever Pinniped reads from session storage, it checks that the version number recorded into the Secret is the same version number that is hardcoded in the Pinniped code. If it is different (e.g. if you upgraded the Supervisor to a version that uses a different session storage version number) then the read will fail and the user will be prompted to log in again. This gives us a simple and graceful way to handle changing the data format of what we store inside those session Secrets. This PR adds a new field to the JSON data stored in those secrets (as shown by the fuzz test), so just to be safe I bumped the storage version numbers.

I also refactored some tests to make it easier to bump the storage version numbers in the future. Future bumps will be result in much smaller test diffs.